### ISSUE TITLE
Mounting a local directory onto itself leads to a hard hang

### ISSUE THREAD

**Issue by estyrke** [ncw/rclone#2905]
#### What is the problem you are having with rclone?

I created a remote with the name OneDrive, and then I used the wrong command to mount it to a directory with the same name:

`rclone mount OneDrive OneDrive`

(The correct command `rclone mount OneDrive: OneDrive` does work)

This led to rclone hanging very hard, not even a `sudo killall -9 rclone` will kill it. 

The mount shows up as ```
local:/home/emil/Skrivbord/Emils/OneDrive on /home/emil/Skrivbord/Emils/OneDrive type fuse.rclone (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
``` and there seems to be no way of getting rid of it apart from rebooting (I tried `fusermount -u OneDrive` and `sudo umount OneDrive`, both fail with "Device or resource busy").

I know I used the wrong syntax, but it would be nice if rclone would detect a circular mount and prevent it from happening rather than locking up. :)

#### What is your rclone version (output from `rclone version`)

```
rclone v1.45
- os/arch: linux/amd64
- go version: go1.11.2
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Ubuntu 18.04, 64 bit

####  Which cloud storage system are you using? (eg Google Drive)

OneDrive (but in fact local file system is being used in this case)

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone mount OneDrive OneDrive`
`ls OneDrive`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
2019/01/10 11:11:01 DEBUG : rclone: Version "v1.45" starting with parameters ["rclone" "-vv" "mount" "OneDrive" "OneDrive"]
Enter configuration password:
password:
2019/01/10 11:11:04 DEBUG : Using config file from "/home/emil/.config/rclone/rclone.conf"
2019/01/10 11:11:04 DEBUG : Local file system at /home/emil/Skrivbord/tmp/OneDrive: Mounting on "OneDrive"
2019/01/10 11:11:04 INFO  : Local file system at /home/emil/Skrivbord/tmp/OneDrive: poll-interval is not supported by this remote
2019/01/10 11:11:04 DEBUG : Adding path "vfs/forget" to remote control registry
2019/01/10 11:11:04 DEBUG : Adding path "vfs/refresh" to remote control registry
2019/01/10 11:11:04 DEBUG : Adding path "vfs/poll-interval" to remote control registry
2019/01/10 11:11:04 DEBUG : : Root: 
2019/01/10 11:11:04 DEBUG : : >Root: node=/, err=<nil>
2019/01/10 11:11:33 DEBUG : /: Attr: 
2019/01/10 11:11:33 DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxr-xr-x, err=<nil>
2019/01/10 11:11:34 DEBUG : /: Attr: 
2019/01/10 11:11:34 DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxr-xr-x, err=<nil>
2019/01/10 11:11:34 DEBUG : /: ReadDirAll: 
^C2019/01/10 11:13:42 Fatal error: failed to umount FUSE fs: exit status 1: fusermount: failed to unmount /home/emil/Skrivbord/tmp/OneDrive: Device or resource busy
```

**Comment by ncw** [comment]
I managed to reproduce this.

Normally `fusermount -z -u mountpoint` sorts this kind of thing out, but not in this case!  The fs did unmount but rclone did not return...

I've fixed this now - the fix will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released with v1.46.

Thanks for reporting - it is a nasty crash.

**Comment by estyrke** [comment]
@ncw Thanks for fixing!
